src/osd/ReplicatedBackend.cc

```diff
@@ -246,12 +246,13 @@ void ReplicatedBackend::on_change()
   dout(10) << __func__ << dendl;
   for (map<ceph_tid_t, InProgressOp>::iterator i = in_progress_ops.begin();
        i != in_progress_ops.end();
-       in_progress_ops.erase(i++)) {
+       i++) {
```

Comment:
might want to use `++i`, as `i` is not a primitive type. will do this in a follow-up PR, though.

Reply:
#19097